### DISCUSSION TITLE
Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node

### DISCUSSION THREAD

**Discussion by darwin-morocho** [vercel/next.js#25049] [Help]
### What version of Next.js are you using?

9.5.5

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start

### Describe the Bug

I'am implementing the facebook messenger chat using this code
```JSX
import { PureComponent } from 'react';

export default class MessengerChat extends PureComponent {
  initialized = false;

  componentDidMount() {
    if (!this.initialized) {
      this.initialized = true;
      const script = document.createElement('script');
      script.src = 'https://connect.facebook.net/es_LA/sdk/xfbml.customerchat.js';
      script.async = true;
      script.onload = () => {
        (window as any).fbAsyncInit = () => {
          (window as any).FB.init({
            xfbml: true,
            version: 'v10.0',
            appId: 220291656074258,
          });
        };
      };
      document.head.appendChild(script);
    }
  }

  render() {
    return (
      <>
        <div id="fb-root" />
        <div className="fb-customerchat" {...{ attribution: 'biz_inbox', page_id: '103155228155970' }} />
      </>
    );
  }
}
```
And importing it as 
```JSX
import dynamic from 'next/dynamic';
const MessengerChat = dynamic(() => import('./MessengerChat'), { ssr: false });
```

The first time when the page loads it works perfect, but when navigate to another page I have the next error
```
DOMException: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.
```
![image](https://user-images.githubusercontent.com/15864336/117864383-58b52780-b25a-11eb-82fe-d6fc268a9550.png)

Using this package https://www.npmjs.com/package/react-messenger-customer-chat I have the same problem

### Expected Behavior

The next page must work without any problem

### To Reproduce

Navigate to pages without refreshing the app

**Comment by ljosberinn** [comment]
doesn't seem related to nextjs

**Comment by jzaburunov** [comment]
@darwin-morocho Hi! Did you solve this?

**Comment by isaacspagnol** [comment]
@darwin-morocho HI! I had the same problem here.. Did you solve it?

**Comment by lyonfuture** [comment]
Any updates on this? Facing the same issue.

**Comment by mAAdhaTTah** [comment]
This is a React error: https://github.com/facebook/react/issues/17256

**Comment by michael-mcgreal** [comment]
Try wrapping the element in an empty div.

Sometimes a library (like google maps) will move the referenced element to a different part of the DOM. So when react tries to remove it, it’s gone. But if you wrap the referenced element in a div, then the div wrapper will remain, even if its child is removed by another non-react process.

**Comment by vishnukumarkvs** [comment]
I recieved this error because of .jsx files. Convert .jsx to js and mark it as a Client Component. Using Nextjs 13.4.3

**Comment by frontendparham** [comment]
if you get this error when you want to create multiple layout.jsx files for different parts of your application make sure to return a react fragment (<></>) from any layout.jsx component Except the root layout. so DO NOT return html and body tag again.

**Reply by Oyinkansolight** [reply]
This was the issue on my end, thank you 🙏

**Reply by khxif** [reply]
thanks it worked for me too..

**Reply by Tabish-binaryvibes** [reply]
Thanks, its worked for me

**Reply by huseyinium** [reply]
Thanks. I removed <body> tags from my page.tsx file and it worked.

**Reply by Retnuh1337** [reply]
This was it, thank you. I removed the html and body elements and replaced them with a react fragment "<></>" in the new layout file I created in side my route folder.

**Reply by Mehalilsafi** [reply]
it work for me to thnx dude

**Reply by mozxmmil** [reply]
thanks bro it's worked

**Reply by jennyliuzhang** [reply]
bless you

**Reply by stevebarakat** [reply]
Thank you!

**Reply by lpodkalicki** [reply]
Thanks! :)

**Reply by Shobhit2806** [reply]
Thanks

**Reply by IxtiyorDeveloper** [reply]
thanks this helped me too!

**Comment by codewithahad01** [comment]
I solution for this error that I had in my code in Next.js
so, basically when making every page I put layout.tsx file for them  and I copied the code for layouts same like root layout that it shouldn't be the same 

```
<html lang="en">
      <body className={edu_tas_beginner.className}>
        {children}
      </body>
  </html>
```

The HTML and body tag should only be in the root layout not in other layouts for pages. so remove the html and body tag for every layout except the root layout.
write below code in every layout files

<div>
          {children}
 </div>

**Reply by yvylol** [reply]
👍 dude, you saved my day.

**Reply by celsolnv** [reply]
👍 dude, you saved my day.

**Reply by shailendrakanherkar18** [reply]
Thanks for the solution.

**Reply by dushesmmm** [reply]
i luv u

**Reply by LollyTim** [reply]
Thanks man this work's

**Reply by JosephMutyaba** [reply]
u saved my day!

**Reply by Oliver-Bagin** [reply]
I had a similar issue. The cause was that I had **accidentally duplicated** the ```<Head> ``` tag, my code looked like this:
```JSX
    <Html>
        <Head>
           <Head>
              <link rel="preconnect" href="https://fonts.googleapis.com" />
            <link
              href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100..900;1,100..900&display=swap"
              rel="stylesheet"
            />
         </Head>
      </Head>
      <body className="antialiased">
        <Main />
        <NextScript />
      </body>
    </Html>
```

Removing the nested ```<Head>``` tag removed this issue. 

Looks like duplicate body and head tags might cause this issue.

**Comment by beforedisappear** [comment]
Any updates? It is big problem when u use google translator

**Comment by 0xshawn** [comment]
same too
